apps/api/src/scraper/WebScraper/crawler.ts

```diff
@@ -64,6 +64,14 @@ export class WebCrawler {
   private filterLinks(sitemapLinks: string[], limit: number, maxDepth: number): string[] {
     return sitemapLinks
       .filter((link) => {
+        
+        // if link is not a complete url, add the base url
+        link = link.trim();
+        const isCompleteUrl = new RegExp('^(?:[a-z+]+:)?//', 'i');
+        if (!isCompleteUrl.test(link)){
+          link = this.baseUrl + link;
+        }
+
         const url = new URL(link);
```

Comment:
```suggestion
        const url = new URL(link.trim(), this.baseUrl);
```

This is easier, faster, and does not rely on the RegEx engine